tests/test_pagination.py

```diff
@@ -13,6 +21,79 @@
 factory = APIRequestFactory()
 
 
+class MockQuerySet:
+    _operator_match = re.compile(r'(?P<field>[a-zA-Z0-9]+)((__)(?P<operator>[a-zA-Z0-9]+))?')
+
+    def __init__(self, items):
+        self.items = items
+        self.model = Mock()
+        self.model._meta.pk.name = 'id'
+
+    def filter(self, q_object=None, **kwargs):
+        if not q_object:
+            q_object = Q(**kwargs)
+
+        query = self._q_object_to_expression(q_object)
+        res = query(self.items[0])
+
+        return MockQuerySet([
+            item for item in self.items
+            if query(item)
+        ])
+
+    def order_by(self, *ordering):
+
+        def _ordering_callable(item):
+            ordering_params = []
+
+            for param in ordering:
+                if param.startswith('-'):
+                    ordering_params.append(0 - getattr(item,param[1:]))
+                else:
+                    ordering_params.append(getattr(item,param))
+
+            return tuple(ordering_params)
+
+        return MockQuerySet(list(sorted(self.items, key=_ordering_callable)))
+
+
+    def __getitem__(self, sliced):
+        return self.items[sliced]
+
+    def _q_object_to_expression(self, q_object):
+        operator_map = {
+            'gt': operator.gt,
+            'lt': operator.lt,
+            'gte': operator.ge,
+            'lte': operator.le,
+            'OR': any,
+            'AND': all,
+        }
+
+        def _parse(_q_object):
+            _statements = []
+
+
+            for child in _q_object.children:
+                if isinstance(child, Q):
+                    return [lambda item: operator_map[child.connector](l(item) for l in _parse(child))]
+
+                match = self._operator_match.match(child[0]).groupdict()
+                field, field_op = match['field'], match['operator']
+
+                if not field_op:
+                    field_op = operator.eq
+                else:
+                    field_op = operator_map[field_op]
+
+                value = child[1]
+                
+                _statements.append(lambda item: field_op(getattr(item, field), int(value)))
+
+            return _statements
+
+        return lambda item: operator_map[q_object.connector](l(item) for l in _parse(q_object))
```

Comment:
I'm not a massive fan of 'MockQuerysets' because we're really reimplementing stuff that's handled by the database.  But I tried to keep the pattern of tests as they are.